doc/10-04-2020-High-Level-Design.md

```diff
@@ -0,0 +1,42 @@
+# Summary
+
+The goal of the RSocket TCK is to allow developers implement RSocket protocol in
+their favoured programming language and be sure, that it follows RSocket 
+specification.
+
+#Motivation
+
+Reactive programming is becoming popular since 2015. That year, the design of 
+RSocket protocol had been started. Protocol pushes reactive semantic to the
+network. Lots of engineers worked on implementations of RSocket protocol in 
+different language. There is no way to check if implementation is compliant with
+protocol specification.
+
+# Detailed Design
+
+To address issues that implementations were compliant with protocol 
+specification we have to introduce a tool (TCK). TCK should use black-box 
+testing against different implementations.
```

Comment:
Here we have to say different. We have to mentioned that TCK designed to have 2 independent parts. 

```suggestion
The TCK designed to have two independent parts. 
The first one is a test runner in the form of a CLI tool along with a form of extendable test suites. The second part is a driver, or simply saying a set of responses to logical interactions which allows ensuring protocol correctness. Please note, that the first part is well specified, where the second part lies on the implementors.
```